NebulaHost/PacketProcessors/Logistics/StationUIBroadcaster.cs

```diff
@@ -22,18 +22,37 @@ public StationUIBroadcaster()
         }
         public void ProcessPacket(StationUI packet, NebulaConnection conn)
         {
-            Player player = playerManager.GetPlayer(conn);
             // if a user adds/removes a ship, drone or warper or changes max power input broadcast to everyone.
-            if (player != null && StationUIManager.UpdateCooldown == 0 &&
+            if (StationUIManager.UpdateCooldown == 0 &&
                 (packet.SettingIndex == StationUI.EUISettings.MaxChargePower
                  || packet.SettingIndex == StationUI.EUISettings.SetDroneCount
                  || packet.SettingIndex == StationUI.EUISettings.SetShipCount
                  || packet.SettingIndex == StationUI.EUISettings.SetWarperCount
-                 || packet.SettingIndex == StationUI.EUISettings.AddOrRemoveItemFromStorageRequest
-                 || packet.SettingIndex == StationUI.EUISettings.AddOrRemoveItemFromStorageResponse)
+                 )
                 )
             {
-                playerManager.SendPacketToAllPlayers(packet);
+                // this is the SendPacketToAllPlayers() logic but we need to set the mimic flag here.
+                using (playerManager.GetConnectedPlayers(out var connectedPlayers))
+                {
+                    foreach (var kvp in connectedPlayers)
```

Comment:
Can use `connectedPlayers.Values` here rather than looping the keypairs

Reply:
That's actually inefficient:
https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1841

Reply:
You allocate a wrapper collection for the values of the dictionary on every access:
https://github.com/Unity-Technologies/mono/blob/e20cf158facacfe732f4b53c0501e7263e114efd/mcs/class/corlib/System.Collections.Generic/Dictionary.cs#L623

Also, you can mistakenly escape `connectedPlayers.Values`, which was not possible before with the local connectedPlayers variable.

Reply:
You might want the key anyway, logging, etc. The fact that there's no logging in the codebase right now is worrying (yes, I know, console logging is slow, allocations in unity's backend, etc, but you don't need to use Unity's logging system.

Reply:
So should i change this or leave it as is?

Reply:
Leave as-is imo.